test/functional/terminal/tui_spec.lua

```diff
@@ -2003,38 +2003,49 @@ describe('TUI', function()
     ]])
   end)
 
-  it('invalidated regions are cleared with terminal background attr', function()
-    local screen = Screen.new(50, 10)
-    screen:set_default_attr_ids({ [1] = { foreground = Screen.colors.Black } })
-    screen:attach()
-    fn.termopen({
-      nvim_prog,
-      '--clean',
-      '--cmd',
-      'set termguicolors',
-      '--cmd',
-      'sleep 10',
-    }, {
-      env = {
-        VIMRUNTIME = os.getenv('VIMRUNTIME'),
-      },
-    })
-    screen:expect({
-      grid = [[
-        {1:^                                                  }|
-        {1:                                                  }|*8
-                                                          |
-      ]],
-    })
-    screen:try_resize(51, 11)
-    screen:expect({
-      grid = [[
-        {1:^                                                   }|
-        {1:                                                   }|*9
-                                                           |
-      ]],
-    })
-  end)
+  for i, guicolors in ipairs({ "notermguicolors", "termguicolors" }) do
+    it('has no black flicker when clearing regions during startup with ' .. guicolors, function()
+      local screen = Screen.new(50, 10)
+      screen:set_default_attr_ids({
+        [1] = { foreground = Screen.colors.Black },
+        [2] = { reverse = true },
+      })
+      screen:attach()
+      fn.termopen({
+        nvim_prog,
+        '--clean',
+        '--cmd',
+        'set ' .. guicolors,
+        '--cmd',
+        'sleep 2',
+      }, {
+        env = {
+          VIMRUNTIME = os.getenv('VIMRUNTIME'),
+        },
+      })
+      screen:expect({
+        grid = i == 1 and [[
+          ^m                                                 |
```

Comment:
Looked at this one more time because I still felt uneasy with some inexplicable debugging I did before. #28676 only worked accidentally for 'termguicolors' because it `normal_bg` is initialized to -1, whereas its cterm counterpart is initialized to 0 and still produced a black screen for the added test here. Again not sure about the validity/usefulness of this test as IDK what ^m means, but it does fail on master where that m dissapears and there is no further difference in the test output.